libusb/core.c

```diff
@@ -2204,6 +2207,111 @@ int API_EXPORTED libusb_set_auto_detach_kernel_driver(
 	return LIBUSB_SUCCESS;
 }
 
+/** \ingroup libusb_dev
+ * Check if the endpoint supports RAW_IO.
+ * \param dev_handle a device handle
+ * \param endpoint the endpoint to check
+ *
+ * \returns 1 if the endpoint supports RAW_IO
+ * \returns 0 if the endpoint does not support RAW_IO
+ * \returns a LIBUSB_ERROR code on failure
+ *
+ * Only endpoints using the WinUSB driver support RAW_IO,
+ * for all other backends/drivers this function will return 0.
+ *
+ * Fails if the interface the endpoint belongs to isn't claimed,
+ * \see libusb_claim_interface().
+ *
+ * \see libusb_endpoint_set_raw_io()
+ *
+ * Since version 1.0.28, \ref LIBUSB_API_VERSION >= 0x0100010B
+ */
+int API_EXPORTED libusb_endpoint_supports_raw_io(libusb_device_handle* dev_handle,
```

Comment:
I know there have been a lot of discussion about the API, and I don't want to delay the merging of this nice PR more than necessary, but I am wondering why is this function needs to be on the public API, given that calling libusb_endpoint_set_raw_io on an endpoint that does not support it will return LIBUSB_ERROR_NOT_SUPPORTED anyway, so it seams that libusb_endpoint_supports_raw_io(dev, ep) is equivalent to libusb_endpoint_set_raw_io(dev, ep, 0) for instance.

I am also wondering about why the choice of 0 and 1 values instead of say, LIBUSB_ERRROR_NOT_SUPPORTED and LIBUSB_SUCCESS for instance.